src/subscription_manager/gui/registergui.py

```diff
@@ -1721,18 +1840,21 @@ def apply(self):
                         "hostname[:port][/prefix]"),
                       None)
             return False
-
-        CFG.save()
+        # Do not persist the new information once we vaildate
+        # wait until after we successfully register
+        #CFG.save()
 
         self.info.set_property('hostname', hostname)
         self.info.set_property('port', port)
         self.info.set_property('prefix', prefix)
 
         if self.activation_key_checkbox.get_active():
+            log.debug('ACTIVATION_PAGE')
```

Comment:
For debugging I think I ended up adding something to RegisterWidget._on_move_to_screen that would log the screen id and it's name (self._screens[screen_id].__class__ maybe?), and usually the contents of 
RegisterWidget.uniq_screen_history and RegisterWidget.applied_screen_history